### PR TITLE
Fix type hints/references.

### PR DESCRIPTION
**Pull request by ndm2**

### CODE REVIEW

File: src/I18n/Time.php [hunk 1]
```diff
@@ -856,10 +856,10 @@ public static function parseTime($time, $format = null)
     /**
      * Convenience method for getting the remaining time from a given time.
      *
-     * @param DateTime $datetime The date to get the remaining time from.
-     * @return DateInterval|bool The DateInterval object representing the difference between the two dates or FALSE on failure.
+     * @param \DateTime $datetime The date to get the remaining time from.
+     * @return \DateInterval|bool The DateInterval object representing the difference between the two dates or FALSE on failure.
      */
-    public static function fromNow($datetime)
+    public static function fromNow(\DateTime $datetime)
```
Comment (ADmad) [inline src/I18n/Time.php]: Please add `use` statement at top to avoid the starting backslash.

Comment (ndm2) [inline src/I18n/Time.php]: Is that the projects standard? There are a few other occurrences of non imported hints, I could fix this all in one go then.

Comment (ADmad) [inline src/I18n/Time.php]: Yes

Comment (ndm2) [inline src/I18n/Time.php]: OK, can I assume that this does apply generally then? ie not only in function/method signatures, but everywhere in the code (`new Abc()`, Abc::xyz(), `catch (Abc)`, ...)?

Comment (ADmad) [inline src/I18n/Time.php]: Everywhere in code.

Comment (ndm2) [inline src/I18n/Time.php]: Alrighty, I'll check everywhere then.

Comment (lorenzo) [inline src/I18n/Time.php]: I world prefer this one not be added, since I often pass a datetimeimmutable object

Comment (ndm2) [inline src/I18n/Time.php]: So maybe `\DateTime|\DateTimeImmutable`?

**Comment by markstory** [comment]
There are a fee phpcs failures that should be fixed.

**Comment by ndm2** [comment]
@markstory Should be fixed... and I noticed that I toally missed out the test sources, do the same "always import" standards apply there?

**Comment by ADmad** [comment]
> do the same "always import" standards apply there?

It should but we allow bit of slackness when it comes to coding standards in tests :stuck_out_tongue:

**Comment by ndm2** [comment]
@ADmad I see, maybe later then :)

@lorenzo **https://github.com/cakephp/cakephp/pull/7440#discussion_r40246657** hasn't been addressed yet, would you be good with `\DateTime|\DateTimeImmutable`?

**Comment by lorenzo** [comment]
@ndm2 `\DateTime|\DateTimeInterface` would do it

**Comment by dereuromark** [comment]
Why not just DateTimeInterface? Both implement it afaik.

**Comment by ADmad** [comment]
Oh, I missed the fact that @lorenzo's comment was not addressed when merging, sorry.

**Comment by ndm2** [comment]
@dereuromark `\DateTimeInterface` is PHP >= 5.5.0. Note sure what the general consensus on stuff like this is, but I thought that it probably wouldn't even be appropriate to use in docblock type hints?

Also the whole `\DateTimeInterface` thingy seems pretty quirky, as `DateTime::diff()` won't accept userland implementations of it, and as of PHP 5.5.8 it's not possible to implement it in userland code anymore at all.

https://wiki.php.net/rfc/drop-datetimeinterface PHP, you gotta love it :smile:

**Comment by ndm2** [comment]
@ADmad No problem, I'll just push it in a new PR :)

**Comment by dereuromark** [comment]
So omitting sounds like the best option then.

**Pull request merged**